site/compiler/test.js

```diff
@@ -133,14 +149,324 @@ describe('site/compiler', () => {
       expect(pages[8].body).to.match(/Upcoming Event/);
       expect(pages[9].path).to.equal('about-us/events/2016/08/03/designing-in-cross-functional-teams/index.html');
       expect(pages[9].body).to.match(/Designing in cross-functional teams/);
     });
+
+    it('renders the dynamic badger pages of the site', () => {
+      const pages = compileSite({
+        jobs: [],
+        job: {},
+        contactUsURL: '',
+        featuredBlogPosts: [],
+        events: [],
+        event: {},
+        badgers: [{
+          firstName: 'Alex',
+          lastName: 'Savin',
+          categories: [
+            'Engineering',
+            'Leadership',
+          ],
+        }],
+        categories: ['Engineering', 'Leadership'],
+      });
+
+      expect(pages.length).to.equal(11);
+      expect(pages[8].path).to.equal('about-us/people/index.html');
+      expect(pages[8].body).to.match(/everyone/);
+      expect(pages[9].path).to.equal('about-us/people/engineering/index.html');
+      expect(pages[9].body).to.match(/engineering/);
+      expect(pages[10].path).to.equal('about-us/people/leadership/index.html');
+      expect(pages[10].body).to.match(/leadership/);
+    });
+
+    describe('with engineer, leadership and pm', () => {
+      it('should render one page of each and everyone', () => {
+        const pages = compileSite({
+          jobs: [],
+          job: {},
+          contactUsURL: '',
+          featuredBlogPosts: [],
+          events: [],
+          event: {},
+          badgers: [
+            {
+              firstName: 'Alex',
+              lastName: 'Savin',
+              categories: [
+                'Engineering',
+                'Leadership',
+              ],
+            },
+            {
+              firstName: 'Sari',
+              lastName: 'Griffiths',
+              categories: [
+                'PM',
+                'Leadership',
+              ],
+            },
+          ],
+          categories: ['Engineering', 'Leadership', 'PM'],
+        });
+
+        expect(pages.length).to.equal(12);
+        expect(pages[8].path).to.equal('about-us/people/index.html');
+        expect(pages[8].body).to.match(/everyone/);
+        expect(pages[9].path).to.equal('about-us/people/engineering/index.html');
+        expect(pages[9].body).to.match(/engineering/);
+        expect(pages[10].path).to.equal('about-us/people/leadership/index.html');
+        expect(pages[10].body).to.match(/leadership/);
+        expect(pages[11].path).to.equal('about-us/people/pm/index.html');
+        expect(pages[11].body).to.match(/pm/);
+      });
+    });
+
+    describe('with one UX & Designer', () => {
+      it('should render one everyone and one ux-design', () => {
+        const pages = compileSite({
+          jobs: [],
+          job: {},
+          contactUsURL: '',
+          featuredBlogPosts: [],
+          events: [],
+          event: {},
```

Comment:
We should define a `baseState` with all of these empty fields then you could just do:

```
{ ...baseState, badgers: mockBadgers }
```

To keep the test data focused.